tests/dhcp_relay/test_dhcp_pkt_fwd.py

```diff
@@ -0,0 +1,329 @@
+         Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            ip(str): IP to add/remove route for
+            peerIp(str): BGP peer IP
+            op(str): operation add/remove to be performed, default add
+
+        Returns:
+            None
+        """
+        logger.info("{0} route to '{1}' via '{2}'".format(
+            "Deleting" if "no" == op else "Adding",
+            ip,
+            peerIp
+        ))
+        duthost.shell("vtysh -c \"configure terminal\" -c \"{} ip route {} {}\"".format(
+            op,
+            ipaddress.ip_interface(unicode(ip + "/24")).network,
+            peerIp
+        ))
+
+    @pytest.fixture(scope="class")
+    def dutPorts(self, duthost, testbed):
+        """
+        Buils list of DUT ports and classify them as Upstream/Downstream ports.
+
+        Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            testbed(Ansible Fixture): testbed information
+
+        Returns:
+            dict: contains downstream/upstream ports information
+        """
+        if testbed["topo"]["name"] not in ("t1", "t1-lag", "t1-64-lag", "t1-64-lag-clet"):
+            pytest.skip("Unsupported topology")
+
+        downstreamPorts = []
+        upstreamPorts = []
+
+        mgFacts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+
+        for dutPort, neigh in mgFacts["minigraph_neighbors"].items():
+            if "T0" in neigh["name"]:
+                downstreamPorts.append(dutPort)
+            elif "T2" in neigh["name"]:
+                upstreamPorts.append(dutPort)
+
+        yield {"upstreamPorts": upstreamPorts, "downstreamPorts": downstreamPorts}
+
+    @pytest.fixture(scope="class")
+    def testPorts(self, duthost, dutPorts):
+        """
+        Select one upstream and one downstream ports for DHCP packet forwarding test
+
+        Args:
+            duthost(Ansible Fixture): instance of SonicHost class of DUT
+            dutPorts(Ansible Fixture, dict): contains downstream/upstream ports information
+
+        Returns:
+            dict: contains downstream/upstream port (or LAG members) information used for test
+        """
+        downstreamLags, downstreamPeerIp = self.__getPortLagsAndPeerIp(
+            duthost,
+            random.choice(dutPorts["downstreamPorts"])
+        )
+        upstreamLags, upstreamPeerIp = self.__getPortLagsAndPeerIp(
+            duthost,
+            random.choice(dutPorts["upstreamPorts"])
+        )
+
+        self.__updateRoute(duthost, self.DHCP_SERVER["ip"], upstreamPeerIp)
+        self.__updateRoute(duthost, self.DHCP_RELAY["ip"], downstreamPeerIp)
+
+        yield {"upstream": upstreamLags, "downstream": downstreamLags}
+
+        self.__updateRoute(duthost, self.DHCP_SERVER["ip"], upstreamPeerIp, "no")
+        self.__updateRoute(duthost, self.DHCP_RELAY["ip"], downstreamPeerIp, "no")
+
+    @classmethod
+    def createDhcpDiscoverRelayedPacket(self, dutMac):
```

Comment:
I admittedly did not look super closely at what their methods are doing, but are we not able to use the different DHCP packet functions in PTF testutils? (example: https://github.com/p4lang/ptf/blob/570ee4c6bb18cf96b0e56685607782ee76b38ca1/src/ptf/testutils.py#L2107)

Reply:
This is more flexible as there are a number of fields that have specific values. Also, we could add other Options here if we want to.